squid/files/squid.conf

```diff
@@ -1,136 +1,84 @@
-# from where browsing should be allowed
-http_access allow localnet
-http_access allow localhost
-{%  endif %}
-
-{%- for acl, ports in cfg_squid.get('port_acl', {}).items() %}
-{%- for port in ports %}
-acl {{acl}} port {{port}}
-{%- endfor %}
-{% endfor %}
-
-{%- for acl, sources in cfg_squid.get('src_acl', {}).items() %}
-{%- for src in sources %}
-acl {{acl}} src {{src}}
-{%- endfor %}
-{% endfor %}
-
-{%- for acl, destinations in cfg_squid.get('dst_acl', {}).items() %}
-{%- for dst in destinations %}
-acl {{acl}} dst {{dst}}
-{%- endfor %}
-{% endfor %}
-
-{%- for acl, domains in cfg_squid.get('dstdomain_acl', {}).items() %}
-{%- for domain in domains %}
-acl {{acl}} dstdomain {{domain}}
-{%- endfor %}
-{% endfor %}
-
-{%- for acl, domains in cfg_squid.get('url_regex_acl', {}).items() %}
-{%- for domain in domains %}
-acl {{acl}} url_regex {{domain}}
-{%- endfor %}
-{% endfor %}
-
-{%- for rule in cfg_squid.get('http_access', []) %}
-http_access {{rule}}
-{%- endfor %}
-
-# And finally deny all other access to this proxy
-http_access deny all
+{%- set http_access = cfg_squid.get('http_access', []) %}
+{{- macros.squid_directive('http_access', http_access) }}
 
-# Squid normally listens to port 3128
-{%- for http_port in cfg_squid.get('http_port', ['3128']) %}
-http_port {{http_port}}
-{%- endfor %}
+{%- set http_port = cfg_squid.get('http_port', []) %}
+{{- macros.squid_directive('http_port', http_port) }}
 
-{%- for https_port in cfg_squid.get('https_port', []) %}
-https_port {{https_port}}
-{%- endfor %}
+{%- set https_port = cfg_squid.get('https_port', []) %}
+{{- macros.squid_directive('https_port', https_port) }}
 
-# Uncomment and adjust the following to add a disk cache directory.
-#cache_dir ufs /var/cache/squid 256 16 256
-{{ cfg_squid.get('cache_dir', '') }}
+{%- set cache_dir = cfg_squid.get('cache_dir', []) %}
+{{- macros.squid_directive('cache_dir', cache_dir) }}
 
 # Leave coredumps in the first cache dir
 coredump_dir {{map.cache}}
 
-#
-# Add any of your own refresh_pattern entries above these.
-#
+# Refresh patterns
+{%- set refresh_patterns = cfg_squid.get('refresh_pattern', []) %}
+{%- if refresh_patterns == 'defaults' %}
 refresh_pattern ^ftp:		1440	20%	10080
 refresh_pattern ^gopher:	1440	0%	1440
 refresh_pattern -i (/cgi-bin/|\?) 0	0%	0
 refresh_pattern .		0	20%	4320
+{%- else %}
+{{- macros.squid_directive('refresh_pattern', refresh_patterns) }}
+{%- endif %}
+
+# cache peer
+{%- set cache_peer = cfg_squid.get('cache_peer', []) %}
+{{- macros.squid_directive('cache_peer', cache_peer) }}
 
-{{ cfg_squid.get('extra_configuration', '') }}
+# cache peer access
+{%- set cache_peer_access = cfg_squid.get('cache_peer_access', []) %}
+{{- macros.squid_directive('cache_peer_access', cache_peer_access) }}
+
+# extra configuration
+{%- set extra_configuration = cfg_squid.get('extra_configuration', []) %}
+{%- for configuration in extra_configuration %}
+{{ configuration }}
+{%- endfor %}
+
```

Comment:
While I understand what you did here, I'm not completely sure changing a 'free-form' `extra_configuration` parameter into a list is such a good idea? It imposes extra burden on the way you need to specify your pillar data (escaping, quoting, etc) and is probably more error-prone? As I said, I'm not completely against nor in favour, just not sure if it provides any real benefit or if it will not make it harder to add blocks of configuration at will.